packages/dev/core/src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -49,40 +49,27 @@ export class DeviceSourceManager implements IDisposable, IObservableManager {
             return null;
         }
 
-        return this._devices[deviceType][deviceSlot];
+        return this._devices[deviceType][deviceSlot] as DeviceSource<T>;
     }
     /**
      * Gets an array of DeviceSource objects for a given device type
-     * @param deviceType Type of Device
+     * @param deviceType - Type of Device
      * @returns All available DeviceSources of a given type
      */
     public getDeviceSources<T extends DeviceType>(deviceType: T): ReadonlyArray<DeviceSource<T>> {
         return this._devices[deviceType].filter((source) => {
             return !!source;
-        });
-    }
-
-    /**
-     * Returns a read-only list of all available devices
-     * @returns All available DeviceSources
-     */
-    public getDevices(): ReadonlyArray<DeviceSource<DeviceType>> {
-        const deviceArray = new Array<DeviceSource<DeviceType>>();
-        for (const deviceSet of this._devices) {
-            deviceArray.push.apply(deviceArray, deviceSet);
-        }
-
-        return deviceArray;
+        }) as Array<DeviceSource<T>>;
     }
 
     /**
      * Default constructor
-     * @param engine Used to get canvas (if applicable)
+     * @param engine - Used to get canvas (if applicable)
      */
     constructor(engine: Engine) {
         const numberOfDeviceTypes = Object.keys(DeviceType).length / 2;
-        this._devices = new Array<Array<DeviceSource<DeviceType>>>(numberOfDeviceTypes);
-        this._firstDevice = new Array<number>(numberOfDeviceTypes);
+        this._devices = new Array(numberOfDeviceTypes);

```

Comment:
Why this change? Isn't this totally un-typesafe?

Reply:
As far as I understand, Typescript should be able to deduce the array's object type from the initial declaration so `<Array<DeviceSource<DeviceType>>>` would be unnecessary.  Is this incorrect?

Reply:
Ah yea ok that makes sense.